### PR TITLE
#148 Cassian Andor, Rebellions Are Built On Hope Unit Implementation

### PR DESCRIPTION
**Pull request by GabeLincoln4**
A test file is also included with the implementation. This is my first pull request to this repository, so I have chosen the Cassian Andor Unit Card.

### CODE REVIEW

File: test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+describe('Cassian Andor, Rebellions Are Built On Hope', function() {
+    integration(function(contextRef) {
+        describe('Cassian Andor\'s triggered ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['cassian-andor#rebellions-are-built-on-hope'],
+                    },
+                    player2: {}
```
Comment (VJubert) [inline test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts:10]: you can omit player2 when he's empty

File: test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                expect(context.cassianAndor.exhausted).toBeTrue();
+            });
+        });
+
+        describe('Cassian Andor\'s triggered ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        resources: ['cassian-andor#rebellions-are-built-on-hope', 'battlefield-marine', 'echo-base-defender', 'tech#source-of-insight', 'daring-raid', 'krayt-dragon', 'atst', 'covert-strength', 'commission', 'hunting-nexu'],
```
Comment (VJubert) [inline test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts:28]: for the next time : you can have only atst on resource for easier copy paste ^^

**Review by VJubert — DISMISSED** [review]

**Comment by AMMayberry1** [comment]
Hi, can you please reach out to me (Veld) on the Karabast discord? We can get you looped in to the dev discussions

File: test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts (hunk 2 above)
Comment (GabeLincoln4) [reply test/server/cards/02_SHD/units/CassianAndorRebellionsAreBuiltOnHope.spec.ts:28]: I see, thank you for letting me know. I will keep that in mind for next time.

**Comment by GabeLincoln4** [comment]
I have update the file with the proposed changes. But for some reason when I tried to make the resources array less than eight items. I get an error message. I have attached a photo of the error message below.
Screenshot 2024-11-23 151358

**Comment by AMMayberry1** [comment]
Also dropping a comment here, let us know in the discord if you're still blocked on the above and we're happy to help debug

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**